utils/model_utils.py

```
@@ -339,18 +344,82 @@ def get_existing_models() -> Generator[Model, None, None]:
+            )
+            training_arguments = TrainingArguments(
+                auto_find_batch_size=True,
+                save_strategy="no",
+                output_dir=output_directory,
+                seed=training_seed,
+                data_seed=data_seed,
+            )
+            trainer = Trainer(
+                model_init=lambda m=model_name, d=dataset: AutoModelForSequenceClassification.from_pretrained(
+                    m,
+                    num_labels=len(dataset.unique("label")),
```

Ruff (0.8.0)

Function definition does not bind loop variable `dataset`

(B023)